### ISSUE TITLE
Accelerate Regular_euclidean_traits::Power_test_3

### ISSUE THREAD

**Issue by afabri** [CGAL/cgal#415]
Replacing the usage of Gmpq with Mpzf should make Mesh_3 faster.
@mglisse Can you give me a hand?

**Comment by mglisse** [comment]
If you profiled and found that the code was spending a lot of time in the Gmpq version of Power_test_3, then yes. In Triangulation_3/include/CGAL/internal/Regular_triangulation_filtered_traits_3.h, you would need to import Exact_kernel_rt and C2E_rt and use those instead of Exact_kernel and C2E in a few Filtered_predicate typedefs below. Now I didn't check what the interface says, if you can put that on top of any kernel, you may need some checks before doing it.
Btw, the last time I heard of Mpzf and Mesh_3, it was Laurent telling me he had to disable it because it was breaking things for a customer...

**Comment by sgiraudot** [comment]
Is this still a feature that we need? Do we know how faster this would make Mesh_3?

**Comment by mglisse** [comment]
This seems strongly related to the "small" feature moving weighted points to the kernel. Once that is done, this will be a 3 character patch :smile: .

**Comment by sgiraudot** [comment]
That's good news :)

**Comment by lrineau** [comment]
I give the PR number, to create cross-links: #585.

**Comment by sloriot** [comment]
@mglisse any chance you can provide the 3 character patch for 4.10 or shall we postpone it to 4.11?

**Comment by mglisse** [comment]
I don't think Power_test_3 exists anymore. Whatever its new name is, just check in interface_macros.h if it is using CGAL_Kernel_pred_RT (then it is already done) or CGAL_Kernel_pred.

**Comment by afabri** [comment]
Power_side_of_oriented_power_sphere

**Comment by mglisse** [comment]
It uses the _RT variant, so I guess this issue is already fixed and can be closed?
